### DISCUSSION TITLE
Is there a reason why query is not being sent to the queryHistogram (metric recording)

### DISCUSSION THREAD

**Discussion by jairomfj** [uptrace/opentelemetry-go-extra#117] [General]
The following piece of code creates an `attrs` array, appends the formatted query and sends the newly created array to the trace. On the other hand, the query is not sent when recording the `queryHistogram`.

```go
	...

	attrs := make([]attribute.KeyValue, 0, len(t.attrs)+1)
	attrs = append(attrs, t.attrs...)
	if query != "" {
		attrs = append(attrs, semconv.DBStatementKey.String(t.formatQuery(query)))
	}

	ctx, span := t.tracer.Start(ctx, spanName,
		trace.WithSpanKind(trace.SpanKindClient),
		trace.WithAttributes(attrs...))
	err := fn(ctx, span)
	span.End()

	if query != "" {
		// t.attrs does not contain the query
		t.queryHistogram.Record(ctx, time.Since(startTime).Milliseconds(), metric.WithAttributes(t.attrs...)) 
	}
```

source: https://github.com/uptrace/opentelemetry-go-extra/blob/main/otelsql/otel.go#L94-L108

**Comment by vmihailenco** [comment]
Because high-cardinality attributes are best to be avoided in metrics.

**Reply by jairomfj** [reply]
Hi @vmihailenco , thank you for the reply! 

I was wondering whether it would make sense to add some formatting for that? Something that would simplify the query to some sort of `OPERATION table`? E.g.:

- `INSERT table`
- `SELECT table`
- etc

I know this won't map all the possible queries depending on the complexity, but for simpler scenarios it could be helpful.

Do you think something like it would make sense?

**Reply by vmihailenco** [reply]
@jairomfj yes, that makes sense, but it is not trivial to extract that information from queries. PR is welcome though :)
